pkgs/by-name/li/libaribcaption/package.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+
+, fontconfig
+, freetype
+
+, CoreFoundation
+, CoreGraphics
+, CoreText
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libaribcaption";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "xqq";
+    repo = "libaribcaption";
+    rev = "v${version}";
+    hash = "sha256-x6l0ZrTktSsqfDLVRXpQtUOruhfc8RF3yT991UVZiKA=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  cmakeFlags = [ "-DBUILD_SHARED_LIBS=ON" ];
+
+  buildInputs =
+    lib.optionals (!stdenv.isDarwin) [ fontconfig freetype ]
+    ++ lib.optionals stdenv.isDarwin [ CoreFoundation CoreGraphics CoreText ];
+
+  meta = with lib; {
+    description = "Portable ARIB STD-B24 Caption Decoder/Renderer";
+    homepage = "https://github.com/xqq/libaribcaption";
+    changelog = "https://github.com/xqq/libaribcaption/releases/tag/${src.rev}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ chayleaf ];
+    platforms = platforms.all;
```

Comment:
I don't think this is accurate. `platforms.all` would include specialized build targets like `javascript-ghcjs` and `wasm64-wasi`

https://github.com/NixOS/nixpkgs/blob/0ac7840a74c53b62f124e86fcde8cdd45f2bc131/lib/systems/doubles.nix#L8-L59

Reply:
`platforms.all` is what fontconfig uses, so I just went with that

Reply:
Unless this purely data processing library explicitly *doesn't* support a certain platform, platforms.all is accurate.

Platforms is about what *could* be supported, not what currently is supported by Nixpkgs packaging of a software.

There's no reason why i.e. `hello` couldn't work on i.e. `wasm64-wasi` for example but it probably doesn't work right now.

Restricting platforms should be done when a package is explicitly only meant for a specific platform such as macOS apps or things that have hard dependencies on Linux-specific libraries.